assets/components/src/plugin-settings/SettingsSection.js

```diff
@@ -0,0 +1,101 @@
+/**
+ * Ads Settings Section.
+ */
+
+/**
+ * WordPress dependencies
+ */
+import { Fragment } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import { ActionCard, Grid, Button, TextControl, CheckboxControl, SelectControl } from '../';
+
+const SettingsSection = ( { settings, disabled, onChange, onUpdate } ) => {
+	const section = settings.find( setting => ! setting.key || setting.key === 'active' );
+	if ( ! section ) {
+		return null;
+	}
+	const activation = settings.find( setting => setting.key === 'active' );
+	const fields = settings.filter( setting => setting.key && setting.key !== 'active' );
+	const getControlComponent = setting => {
+		if ( Array.isArray( setting.options ) && setting.options.length ) {
+			return SelectControl;
+		}
+		switch ( setting.type ) {
+			case 'checkbox':
+			case 'boolean':
+				return CheckboxControl;
+			default:
+				return TextControl;
+		}
+	};
+	const getControlType = setting => {
+		switch ( setting.type ) {
+			case 'int':
+			case 'integer':
+			case 'float':
+			case 'number':
+				return 'number';
+			case 'string':
+			case 'text':
+				return 'text';
+			default:
+				return null;
+		}
+	};
+	const getControlProps = setting => ( {
+		disabled,
+		name: `${ setting.section }_${ setting.key }`,
+		type: getControlType( setting ),
+		label: setting.description,
+		help: setting.help || null,
+		options:
+			setting.options?.map( option => ( {
+				value: option.value,
+				label: option.name,
+			} ) ) || null,
+		value: setting.value,
+		checked: setting.type === 'boolean' ? !! setting.value : null,
+		onChange: value => {
+			onChange( setting.key, value );
+		},
+	} );
+	return (
+		<ActionCard
+			isMedium
+			disabled={ disabled }
+			title={ section.description }
+			description={ section.help }
+			toggleChecked={ activation ? activation.value : null }
+			hasGreyHeader={ !! activation }
+			toggleOnChange={ value => onUpdate( { [ activation.key ]: value } ) }
+		>
+			{ ( ! activation || activation.value ) && (
+				<Fragment>
+					<Grid columns={ fields.length === 3 ? 3 : 2 } gutter={ 32 }>
```

Comment:
Suggestion: with a single field or with six fields two columns will be rendered (while it seems one and three would be intended). Modulo should rather be used (but only for >1 item counts.)